api/data_explorer/__main__.py

```diff
@@ -199,26 +207,43 @@ def _process_facets(es):
     app.app.config['NESTED_PATHS'] = elasticsearch_util.get_nested_paths(es)
 
     for facet_config in facets_config:
-        es_field_name = facet_config['elasticsearch_field_name']
-        if es_field_name in facets:
-            raise EnvironmentError('%s appears more than once in ui.json' %
-                                   es_field_name)
-        field_type = elasticsearch_util.get_field_type(es, es_field_name)
-        ui_facet_name = facet_config['ui_facet_name']
-        if es_field_name.startswith('samples.'):
-            ui_facet_name = '%s (samples)' % ui_facet_name
-
-        facets[es_field_name] = {
-            'ui_facet_name': ui_facet_name,
-            'type': field_type
-        }
-        if 'ui_facet_description' in facet_config:
-            facets[es_field_name]['description'] = facet_config[
-                'ui_facet_description']
-
-        facets[es_field_name][
-            'es_facet'] = elasticsearch_util.get_elasticsearch_facet(
-                es, es_field_name, field_type)
+        es_base_field_name = facet_config['elasticsearch_field_name']
+        if elasticsearch_util.is_time_series(es, es_base_field_name):
```

Comment:
I would move the time_series specific code to its own new function.

Something like

is_time_series, time_series_vals, es_field_names = get_time_series_info()